test/libsolidity/semanticTests/payable/no_nonpayable_circumvention_by_modifier.sol

```diff
@@ -0,0 +1,17 @@
+contract C {
+    modifier tryCircumvent {
+        if (false) _; // avoid the function, we should still not accept ether
+    }
+    function f() tryCircumvent public returns (uint) {
+        return msgvalue();
+    }
+    function msgvalue() internal returns (uint) {
+        return msg.value;
+    }
+    function balance() external returns (uint) {
```

Comment:
See #10426.

We could keep this test in `SolidityEndToEndTest` or move it out with this helper. I would not wait for #10426 to be implemented though.

Reply:
Should this commit be kept which extracts it to semantictests or keep the old endtoendtest?

Reply:
I think it's not the only place where tests use workarounds. I've seen tests that include `assert()` to do checks or just have extra functions that check stuff that's not easy (or convenient) to check explicitly.

This preserves the functionality so I'd be for keeping it extracted and just adding and note with a link to this PR in #10426 to remove the helper when the feature is implemented.

Reply:
Left a TODO in the file too.